sycl/include/sycl/ext/intel/esimd/detail/util.hpp

```diff
@@ -183,7 +183,8 @@ template <unsigned N> class ForHelper {
 /// the byte offset \p Offset.
 template <typename T, typename AccessorTy>
 T *accessorToPointer(AccessorTy Acc, uint32_t Offset = 0) {
-  auto BytePtr = reinterpret_cast<char *>(Acc.get_pointer().get()) + Offset;
+  auto BytePtr =
+      reinterpret_cast<char *>(const_cast<T *>(Acc.get_pointer())) + Offset;
```

Comment:
Unless I am misunderstanding the intention of this function, I don't believe there's a guarantee that `T` is the same as the type inside `Acc`. I would however assume that if `get_pointer` returns a const pointer (either the accessor is read-only or the data type of the accessor is const qualified), `T` would also be const. @v-klochkov - Is that assumption correct?

Using the assumption above, I think it would be better to change this to something along the lines of:
```suggestion
  using QualCharPtrType = std::conditional_t<std::is_const_v<AccessorTy::value_type>, const char *, char *>;
  auto BytePtr = reinterpret_cast<QualCharPtrType>(Acc.get_pointer()) + Offset;
```

Reply:
Thanks @steffenlarsen 
I thought of above, I chose **const_cast** 1) to conform with updates in multi_ptr and ii) was thinking of potential scenario, which is described why I went for using **const_cast** in multi_ptr.

Reply:
In the end I suspect we would want to avoid using `const_cast` in the `multi_ptr` implementation too (see my comments on there). If we can avoid it here too, that would definitely be preferable too.

Reply:
I spent two days last week to avoid using `const_cast`, knowing it would be controversial.
I give it further try  :crossed_fingers: